controllers/namespace_controller.go

```diff
@@ -75,14 +75,39 @@ func (r *NamespaceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		return ctrl.Result{}, err
 	}
 
+	log.Log.Info("Processing namespace " + namespace.Name)
+
 	// Detect annotation
-	labelIsPresent := len(namespace.Annotations[namespaceCapabilityNameLabel]) > 0
+	getCrossplaneEnabledAnnotationName, err := getAnnotationEnvVarName("DFDS_CROSSPLANE_ENABLED_ANNOTATION_NAME", "dfds-crossplane-enabled")
+	if err != nil {
+		log.Log.Error(err, "unable to get Crossplane Enabled annotation name")
+	}
+
+	getAWSAccountIDAnnotationName, err := getAnnotationEnvVarName("DFDS_CROSSPLANE_AWS_ACCOUNT_ID_ANNOTATION_NAME", "dfds-aws-account-id")
+	if err != nil {
+		log.Log.Error(err, "unable to get AWS Account annotation name")
+	}
+
+	crossplaneEnabled := "false"
+	if len(namespace.Annotations[getCrossplaneEnabledAnnotationName]) > 0 {
+		log.Log.Info("Crossplane annotation value found: " + namespace.Annotations[getCrossplaneEnabledAnnotationName])
+		crossplaneEnabled = namespace.Annotations[getCrossplaneEnabledAnnotationName]
+	}
 
-	clusterRole := &rbac.ClusterRole{
+	awsAccountId := ""
+	if len(namespace.Annotations[getAWSAccountIDAnnotationName]) > 0 {
+		log.Log.Info("AWS Account ID value found: " + namespace.Annotations[getAWSAccountIDAnnotationName])
+		awsAccountId = namespace.Annotations[getAWSAccountIDAnnotationName]
+	}
+
+	//configMap := getConfigMap("DFDS_CROSSPLANE_CONFIGMAP_NAME", "DFDS_CROSSPLANE_CONFIGMAP_NAMESPACE")
+
+	// TODO: Obtain this clusterrole info from a Configmap or other source, rather than hard code
+	clusterRole := &rbacv1.ClusterRole{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: "providerconfig-" + namespace.Name,
```

Comment:
should we use this format <cloudprovider>-namespace ? example aws-my-namespace

Reply:
Makes sense, I have changed it